### PR TITLE
adding new top level azure_arm gem for ARM

### PR DESCRIPTION
**Pull request by vishrutshah**
This prototype creates an new `azure_arm` gem that includes all other Azure Resource Management gem as dependency so that installing `azure_arm` gem users can get all the arm gems rather than installing them one at a time. 

Updated the rake 
`arm:build` - task to build new azure_arm.gemspec along with other gems
`arm:clean` - task to remove azure_arm<version>.gem along with other gems
`arm:release` - task to push azure_arm gem and tag git repo

Updated REGEN_METADATA to read version from central place (ARM_VERSION) while regenerating library using autorest.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@vishrutshah__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

File: resource_management/Rakefile [hunk 1]
```diff
@@ -106,102 +112,107 @@ def each_gem
...
     autorest_loc: '../../../autorest/binaries/net45/AutoRest.exe',
+    azure_arm: {
+        version: version,
+        tag: 'azure_arm'
+    },
     azure_mgmt_authorization: {
         spec_uri: 'https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/arm-authorization/2015-07-01/swagger/authorization.json',
         ns: 'Azure::ARM::Authorization',
-        version: '0.2.1',
+        version: version,
```
Comment (veronicagg) [inline resource_management/Rakefile]: could this be Azure::ARM::Authorization::VERSION? I'm not sure if all gems will always rev at the same time. It's fine for the current case, so we may end up modifying this later anyway.

File: resource_management/azure_arm.gemspec [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+# Licensed under the MIT License. See License.txt in the project root for license information.
+
+version = File.read(File.expand_path('../ARM_VERSION', __FILE__)).strip
+
+Gem::Specification.new do |spec|
+  spec.name          = 'azure_arm'
+  spec.version       = version
+  spec.authors       = 'Microsoft Corporation'
+  spec.email         = 'azrubyteam@microsoft.com'
+  spec.description   = 'Microsoft Azure Web Apps Management Client Library for Ruby'
```
Comment (veronicagg) [inline resource_management/azure_arm.gemspec]: needs update to text?

File: resource_management/azure_arm.gemspec [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  spec.summary       = 'Official Ruby client library to consume Microsoft Azure Web Apps Management services.'
```
Comment (veronicagg) [inline resource_management/azure_arm.gemspec]: needs update to text?

File: resource_management/azure_arm.gemspec [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  spec.license       = 'MIT'
+
+  spec.required_ruby_version = '>= 1.9.3'
+
+  spec.add_development_dependency 'bundler', '~> 1.9'
+  spec.add_development_dependency 'rake', '~> 10'
+  spec.add_development_dependency 'rspec', '~> 3'
+  spec.add_development_dependency 'dotenv', '~> 2'
+
+  spec.add_runtime_dependency 'azure_mgmt_authorization', version
```
Comment (veronicagg) [inline resource_management/azure_arm.gemspec]: per my comment above, would version here match all the time?

Comment (veronicagg) [inline resource_management/azure_arm.gemspec]: should we be adding the dependency to a specific version or use the pessimistic operator?

**Comment by veronicagg** [comment]
@vishrutshah I left some comments, rest looks good. Thanks!

**Comment by vishrutshah** [comment]
@devigned & @veronicagg 

Let's chat about how we want to manage versioning of each service gem and top level gem.

**Comment by vishrutshah** [comment]
Closing this PR w.r.t PR @ [https://github.com/Azure/azure-sdk-ruby/pull/2](https://github.com/Azure/azure-sdk-ruby/pull/2)

**Pull request closed without merging**